sbi/inference/trainers/base.py

```diff
@@ -306,6 +320,193 @@ def train(
     ) -> NeuralPosterior:
         raise NotImplementedError
 
+    @abstractmethod
+    def _get_potential_function(
+        self,
+        prior: Distribution,
+        estimator: Union[RatioEstimator, ConditionalEstimator],
+    ) -> Tuple[BasePotential, TorchTransform]:
+        """Subclass-specific potential creation"""
+        pass
+
+    def build_posterior(
+        self,
+        estimator: Optional[Union[RatioEstimator, ConditionalEstimator]],
+        prior: Optional[Distribution],
+        sample_with: Literal[
+            "mcmc", "rejection", "vi", "importance", "direct", "sde", "ode"
+        ],
+        **kwargs,
+    ) -> NeuralPosterior:
+        r"""Method for building posteriors.
+
+        This method serves as a base method for constructing a posterior based
+        on a given estimator and prior. The posterior can be sampled using one of
+        several inference methods specified by `sample_with`.
+
+        Args:
+            estimator: The estimator that the posterior is based on.
+            prior: A probability distribution that expresses prior knowledge about the
+                parameters, e.g. which ranges are meaningful for them. Must be a PyTorch
+                distribution, see FAQ for details on how to use custom distributions.
+            sample_with: The inference method to use. Must be one of:
+                - "mcmc"
+                - "rejection"
+                - "vi"
+                - "importance"
+                - "direct"
+                - "sde"
+                - "ode"
+            **kwargs: Additional method-specific parameters.
+
+        Returns:
+            NeuralPosterior object.
+        """
+
+        if prior is None:
+            cls_name = self.__class__.__name__
+            assert (
```

Comment:
You are using assert statements for error handling related to inputs. Getting an `AssertionError` is not super informative for users and is usually only used for things that are super unlikely or almost impossible to happen. I would prefer `ValueErrors` or even custom exceptions.